nevergrad/optimization/optimizerlib.py

```diff
@@ -59,6 +59,9 @@ def __init__(
     ) -> None:
         super().__init__(parametrization, budget=budget, num_workers=num_workers)
         self._sigma: float = 1
+        all_params = paramhelpers.flatten_parameter(self.parametrization)
+        arity = max(len(param.choices) if isinstance(param, p.TransitionChoice) else 500 for param in all_params.values())
+        self.arity_for_discrete_mutation = arity
```

Comment:
just to make sure it's clear: this is the maximum arity of any sub-parameter in the data, but it does not mean the whole data is discrete

Reply:
Yes. Otherwise we use an arbitrary somehow large number.